tools/pubsys/src/aws/ami/register/mk_amispec.rs

```
@@ -0,0 +1,833 @@
+    pub ami: AmiRenderContext,
+    pub block_devices: HashMap<String, BlockDeviceRenderContext>,
+}
+
+impl AmispecRenderContext {
+    /// Create the handlebars render context that is used to render the amispec template
+    fn from_ami_args(
+        ami_args: &AmiArgs,
+        bottlerocket_snapshots: Option<&BottlerocketSnapshots>,
+    ) -> Result<Self> {
+        Ok(AmispecRenderContext {
+            ami: AmiRenderContext {
+                unique_name: ami_args.name.clone(),
+                arch: ami_args.arch.clone(),
+                description: ami_args.description.clone(),
+            },
+            block_devices: bottlerocket_snapshots
+                .map(|bottlerocket_snapshots| {
+                    let image_layout = ami_args.variant_manifest.image_layout().context(
+                        amispec_error::MissingImageLayoutSnafu {
+                            variant_manifest: Box::new(ami_args.variant_manifest.clone()),
+                        },
+                    )?;
+                    Ok(bottlerocket_snapshots.block_device_render_context(image_layout))
+                })
+                .transpose()?
+                .unwrap_or_default(),
+        })
+    }
+}
+
+#[derive(Debug, Clone, Serialize, PartialEq, Eq)]
+struct AmiRenderContext {
+    pub unique_name: String,
+    pub arch: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub description: Option<String>,
+}
+
+#[derive(Debug, Clone, Serialize, PartialEq, Eq)]
+struct BlockDeviceRenderContext {
+    pub device_name: String,
+    pub snapshot_id: String,
+    pub volume_size: u64,
+}
+
+impl BottlerocketSnapshots {
+    /// Creates the block device handlebars render context from a set of created EBS snapshots
+    fn block_device_render_context(
+        &self,
+        image_layout: &ImageLayout,
+    ) -> HashMap<String, BlockDeviceRenderContext> {
+        let (os_volume_size, data_volume_size) = image_layout.publish_image_sizes_gib();
```

It seems like we could additionally validate the partition plan from the image layout; if it's "split" and we only have one snapshot, then we should throw an error?